src/Utilities/UnpackFields/UnpackFields.jl

```diff
@@ -0,0 +1,45 @@
+module UnpackFields
+
+export @unpack_fields
+
+"""
+    unpack_fields(obj, syms...)
+
+Unpack struct properties `syms`
+from `struct` or `NamedTuple` `obj`
+
+# Example
+```julia
+julia> struct Foo
+           a
+           b
+           c
+       end
+
+julia> f = Foo(1,2,3)
+Foo(1, 2, 3)
+
+julia> @unpack_fields f a c
+
+julia> @show a
+a = 1
+1
+
+julia> @show c
+c = 3
+3
+```
+"""
+macro unpack_fields(obj, syms...)
+    thunk = Expr(:block)
+    for sym in syms
+        push!(
+            thunk.args,
+            :($(esc(sym)) = getproperty($(esc(obj)), $(QuoteNode(sym)))),
+        )
+    end
+    push!(thunk.args, nothing)
+    return thunk
```

Comment:
Alternatively:
```suggestion
    exs = [
        :($(esc(sym)) = getproperty($(esc(obj)), $(QuoteNode(sym))))
        for sym in syms
    ]
    Expr(:block, exs...)
```
But what you have is perhaps clearer.

Reply:
Hm, good point. Let's go with your suggestion.